src/plugins/content_management/public/registry/registry.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { ContentTypeDetails } from './types';
+import { ContentType } from './content_type';
+
+export class ContentRegistry {
```

Comment:
I wonder why we need the registry client side. Couldn't we have this on the server and a method (cached) to fetch once those details from the server?

Reply:
Good question. I'll put this question here https://docs.google.com/document/d/13aOspEx_yGcdaRwXZTAOCaiPuAIZlcrkZopvR0cr2_k/edit# and will iterate on it